core/controllers/profile.py

```diff
@@ -454,7 +454,7 @@ def get(self):
         else:
             if self.request and self.request.get('current_url'):
                 target_url = (
-                    '/' if self.request.get('current_url').endswith(
+                    '/splash' if self.request.get('current_url').endswith(
```

Comment:
Shouldn't this code change, based on our previous discussion? I don't see the need for special-casing /splash any more.

Reply:
Actually, no. As I mentioned [here](https://github.com/oppia/oppia/pull/9109/files?file-filters%5B%5D=.html&file-filters%5B%5D=.js&file-filters%5B%5D=.py&file-filters%5B%5D=.ts#r410740304), we need to replace `/` with `/splash` since now the redirection to the default dashboard after logging in and upon clicking the logo would be done via `/splash`. It was being done by `/` previously, but that's in the front now, so we can't use it which is why we use `/splash` instead.

Reply:
/splash seems like an odd choice for the URL of a page whose purpose is to redirect to your "home dashboard". So some changes should probably be made here, but I'm not sure exactly which ones.

More specifically, I don't quite understand what you mean by "/ is in the front now" and why "/" can't handle the redirection. Wouldn't "/" go to the correct place anyway after you've logged in?

Reply:
By "`/` is in the front now", I mean that the roles have kinda been reversed. The way I understand things worked is that previously when you typed in `/splash` in the URL bar, you'd be taken to the splash page no matter what. And, to handle dynamic redirection to the default dashboard based on login status, `/` was being used (by this profile get function and the logo URL). So now, since we don't want the word 'splash' in the URL, I've simply reversed the roles, i.e., `/` for the splash page and `/splash` for the dynamic redirection. I can rename `/splash` to `/home` if that helps. Does that seem acceptable?